### PR TITLE
florist: init at 24.2

### PR DESCRIPTION
**Pull request by lutzberger**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fl/florist/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  lib,
+}:
+
+stdenv.mkDerivation rec {
+  name = "florist";
+  version = "24.2";
+  src = fetchFromGitHub {
+    owner = "adacore";
+    repo = "florist";
+    rev = "refs/heads/${version}";
```
Comment (Aleksanaa) [inline pkgs/by-name/fl/florist/package.nix:15]:
```suggestion
    rev = "refs/tags/${version}";
```

File: pkgs/by-name/fl/florist/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  stdenv,
+  gnat13,
+  gnat13Packages,
+  fetchFromGitHub,
+  lib,
+}:
+
+stdenv.mkDerivation rec {
+  name = "florist";
+  version = "24.2";
```
Comment (Aleksanaa) [inline pkgs/by-name/fl/florist/package.nix:11]:
```suggestion
  version = "24.2";

```

File: pkgs/by-name/fl/florist/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+
+  meta = {
+    description = "Posix Ada Bindings";
+    homepage = "https://github.com/adacore/florist";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ lutzberger ];
+    platforms = lib.platforms.linux;
+  };
+
```
Comment (Aleksanaa) [inline pkgs/by-name/fl/florist/package.nix:35]:
```suggestion
```

File: pkgs/by-name/fl/florist/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner = "adacore";
+    repo = "florist";
+    rev = "refs/heads/${version}";
+    hash = "sha256-EFGmcQfWpxEWfsAoQrHegTlizl6siE8obKx+fCpVwUQ=";
+  };
+
+  configurePhase = ''
+    ./configure --enable-shared --prefix=$out
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/fl/florist/package.nix:21]:
```suggestion
  configureFlags = [ "--enable-shared" ];
```

File: pkgs/by-name/fl/florist/package.nix (hunk 1 above)
Comment (lutzberger) [reply pkgs/by-name/fl/florist/package.nix:16]: The repository has only one tag v20.2, which refers to an older unstable version. Branches are maintained instead, and the latest stable is 24.2. That was the reason, why I have used "heads".

**Comment by lutzberger** [comment]
This is still not merged, I have no authorisation, who can do this?

**Review by Aleksanaa — APPROVED** [review]
Please squash all commits into `florist: init at 24.2`.

**Comment by lutzberger** [comment]
Oh yes, I forgot, Now its done.

**Pull request merged**